pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -645,8 +645,18 @@ func resolveTask(
 			case errors.Is(err, remote.ErrRequestInProgress):
 				return rt, err
 			case err != nil:
+				name := pipelineTask.TaskRef.Name
```

Comment:
so if you look a few lines up @khrm @savitaashture at line 643:

`t, _, vr, err := getTask(ctx, pipelineTask.TaskRef.Name)`

I am unclear on how this would ever work if `pipelineTask.TaskRef.Name` is an empty string.  Maybe the name from the param is seeded in the `getTask` func when it is constructed and the `pipelineTask.TaskRef.Name` is ignored?  I did not have time to trace through the code and find the image bundle version of `getTask` to sort that out.

I'm going to try to nail that down today.

Reply:
OK the amount of layering can make one dizzy, but *eventually* we land at https://github.com/tektoncd/pipeline/blob/main/pkg/resolution/resolver/bundle/params.go#L79 and https://github.com/tektoncd/pipeline/blob/main/pkg/resolution/resolver/bundle/bundle.go#L112

it uses the params and ignore the name for image bundles

I'm going to try and see about using  `OptionsForParams` over in https://github.com/tektoncd/pipeline/blob/main/pkg/resolution/resolver/bundle/params.go#L43

Reply:
wait .... parms can vary by resolution type .... need to research more